s29862-JuliaButkiewicz/zajecia3/zadanie1.txt

```diff
@@ -0,0 +1,72 @@
+Zadanie1
+1.Napisz skrypt, który sprawdzi, czy podano więcej niż jeden parametr.
+Jeśli tak to niech wyświetli pierwszy parametr. Jeśli nie, to niech
+wyświetli komunikat, że nie podano parametrów.
+
+Odpowiedź:
+#!/bin/bash
+
+if [ $# -gt 1 ]; then
+	echo "Pierwszy parametr: $1"
+else
+	echo "Jest tylko jeden parametr: $1"
+fi
+
+2.Napisz skrypt który:
+-Sprawdzi, czy istnieje plik config.cfg w bieżącym katalogu.
+-Jeśli taki plik istnieje, to niech go wykona.
+-Jeśli nie istnieje, to niech sprawdzi, czy podano argument i wykona plik
+o nazwie takiej jak wartość argumentu skryptu.
+-Jeśli żadne z powyższych nie nastąpiło, to niech wyświetli komunikat
+o błędzie.
+
+Odpowiedź:
+#!/bin/bash
+
+if [ -f "config.cfg" ]; then
+	echo "Wykonuje..."
+	./config.cfg
+elif [ $# -eq 1 ]; then
+	name="$1"
+
+	if [ -f "$name" ]; then
+		echo "Wykonywanie pliku $name..."
+		./"$name"
+	else
+		echo "Plik $name nie istnieje."
+	fi
+else
+	echo "BŁĄD!!! Nie ma pliku config.cfg"
+fi
+
+3.Napisz skrypt, który:
+-Sprawdzi, czy jego nazwa kończy się na .sh .
+-Jeśli nie, to niech zmieni swoją nazwę poprzez dodanie rozszerzenia.
+
+Odpowiedź:
+#!/bin/bash
+
+if [[ "$0" =~ \.sh$ ]]; then
+	echo "Nazwa kończy się na .sh"
+else
+	echo "Nazwa nie kończy się na .sh"
+	mv "$0" "$0.sh"
+	echo "Nazwa zmieniona na $0.sh"
+	exit 0
+fi
+
+Zadanie4.
+Napisz skrypt który:
+-Sprawdzi czy w bieżącym katalogu jest więcej niż 5 plików.
+-Jeśli tak, to wypisze odpowiedni komunikat z informcją że tak jest.
+
+Odpowiedź:
+#!/bin/bash
+
+liczba=$(ls -1 | wc -l)
```

Comment:
To jest do poprawki. Jeśli w aktualnym katalogu będą jakieś katalogi to ten skrypt będzie działał błędnie.